include/nuttx/compiler.h

```diff
@@ -255,15 +255,6 @@
 #  undef  CONFIG_PTR_IS_NOT_INT
 #endif
 
-/* GCC supports inlined functions for C++ and for C version C99 and above */
-
-#  if defined(__cplusplus) || (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L)
```

Comment:
@patacongo does this change kill the retro computing support?

Reply:
This states that inline is not just a cpp @xiaoxiang781216 Did you ask @patacongo why this was abstracted the way it was?

Reply:
Since inline doesn't work well in the old compiler, you can see each place which reference CONFIG_HAVE_INLINE have to provide two versions: one is inline function and another is macro which make the simple thing complex(@yamt and I have the same feeling). The better approach is:
1.Remove CONFIG_HAVE_INLINE
2.Implement the function as a normal(no inline) function as much as possible
3.For the simple and performance critical function:
  a.Implement as the macro if it is an internal function
  b.C standard function(like isspace) provide two versions:
     1.Inline version if __cpluscplus__ define
     2.Macro verson if __cpluscplus__ not define
Note: 3.b is required because the c++ library often write "using ::isspace" which is incompatible with macro.